kore/src/Log.hs

```diff
@@ -233,18 +227,21 @@ instance MonadLog log => MonadLog (ReaderT a log)
 
 instance MonadLog log => MonadLog (Strict.StateT state log)
 
-data LoggerEnv monad =
-    LoggerEnv
-        { logAction :: !(LogAction monad ActualEntry)
-        , context :: !(Seq SomeEntry)
-        }
-    deriving (GHC.Generic)
+-- ---------------------------------------------------------------------
+-- * LoggerT
 
 newtype LoggerT m a =
     LoggerT { getLoggerT :: ReaderT (LoggerEnv m) m a }
     deriving (Functor, Applicative, Monad)
     deriving (MonadIO, MonadThrow, MonadCatch, MonadMask)
 
+data LoggerEnv monad =
+    LoggerEnv
+        { logAction :: !(LogAction monad ActualEntry)
+        , context :: ![SomeEntry]
```

Comment:
Why is the context a list now?

Reply:
The list has better sharing. When we add an entry to the context, the `tail` of the list is always the old context. `Seq` uses a tree internally, so every 2-3 new entries, it allocates a new branch of the tree which isn't shared.